shared/domains/chelonia/utils.js

```diff
@@ -547,7 +547,7 @@ export const recreateEvent = (entry: SPMessage, state: Object, contractsState: O
   const newOp = [opT, newRawOpV]
 
   entry = SPMessage.cloneWith(
-    head, newOp, { previousHEAD, height: previousHeight + 1 }
+    head, newOp, { previousKeyOp, previousHEAD, height: previousHeight + 1 }
```

Comment:
I'm not seeing where in this PR you are setting the `SPMessage`'s `_head.previousKeyOp`... Above in `chelonia.js` you do not do it, and it needs to be done in `createV1_0` (which has been updated to receive it, but this is never used)

Reply:
It's set in utils.js:

```js
  entry = SPMessage.cloneWith(
    head, newOp, { previousKeyOp, previousHEAD, height: previousHeight + 1 }
  )
```

It's not set in `chelonia.js` for the same reason the `height` isn't set: we need to fetch the latest head info to find out.